### ISSUE TITLE
Improve Misleading Parse Errors

### ISSUE THREAD

**Issue by DJRHails** [flintlang/flint#246]
```
Expected token '}' at line 10, column 3:
  func foo() -> Address {
  ^^^^
Failed to compile.
```

If there is a malformed expression in the parser it currently outputs an unhelpful error of expecting a closing token. 

This is because the code attempts all possible options for its body and all the errors are hidden from the user.
